plugins/Marketplace/PackageModel.py

```diff
@@ -1,16 +1,25 @@
 # Copyright (c) 2021 Ultimaker B.V.
 # Cura is released under the terms of the LGPLv3 or higher.
 
-from PyQt5.QtCore import pyqtProperty, QObject
 import re
+from enum import Enum
 from typing import Any, Dict, List, Optional
 
+from PyQt5.QtCore import pyqtProperty, QObject, pyqtSignal
+
+from cura.CuraApplication import CuraApplication
 from cura.Settings.CuraContainerRegistry import CuraContainerRegistry  # To get names of materials we're compatible with.
 from UM.i18n import i18nCatalog  # To translate placeholder names if data is not present.
 
 catalog = i18nCatalog("cura")
 
 
+class ManageState(Enum):
```

Comment:
Needs documentation. What do these states mean? Especially halted has a special connotation in this context